ClothesApp/WebAPI/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,101 @@
+using Application.Dtos.Products;
+using Application.Dtos.Reviews;
+using Application.Interfaces.Services;
+using Domain.Entities;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebAPI.Controllers;
+
+[ApiController]
+[Route("api/products")] 
+public class ProductsController : ControllerBase
+{
+    private readonly IProductService _productService;
+
+    private readonly IReviewService _reviewService;
+
+    public ProductsController(IProductService productService, IReviewService reviewService)
+    {
+        _productService = productService;
+        _reviewService = reviewService;
+    }
+    
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IList<ProductDto>))]
+    public async Task<ActionResult> GetProducts()
+    {
+        var products = await _productService.GetAll();
+
+        return Ok(products);
+    }
+    
+    [HttpGet("{id}/is-available")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(string))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ProblemDetails))]
+    public async Task<ActionResult> CheckProductsAvailability([FromRoute] long id)
```

Comment:
Why do you return strings instead of bool values and why would you even need this endpoint? If you have get endpoint for list of products and each product includes quantity, there is no point to make another method that simply says if product is available, you can understand this by quantity